src/chocolatey/infrastructure.app/commands/ChocolateyCacheCommand.cs

```diff
@@ -0,0 +1,307 @@
+                return;
+            }
+
+            var expirationTimer = GetCacheExpiration(configuration);
+
+            var filesBeforeClean = _fileSystem.GetFiles(cacheLocation, "*.dat", SearchOption.AllDirectories);
+
+            if (configuration.CacheCommand.RemoveExpiredItemsOnly)
+            {
+                filesBeforeClean = filesBeforeClean.Where(f => _fileSystem.GetFileModifiedDate(f) < expirationTimer);
+            }
+
+            var beforeFilesCount = filesBeforeClean.Count();
+
+            if (beforeFilesCount == 0)
+            {
+                this.Log().Info("No cached items available to be removed in '{0}'.", cacheLocation);
+                return;
+            }
+
+            if (configuration.CacheCommand.RemoveExpiredItemsOnly)
+            {
+                // We need to remove each individual file when the user only request
+                // deleting expired items. This takes a bit longer.
+                foreach (var fileToRemove in filesBeforeClean)
+                {
+                    _fileSystem.DeleteFile(fileToRemove);
+                }
+
+                foreach (var directoryToRemove in _fileSystem.GetDirectories(cacheLocation))
+                {
+                    if (!_fileSystem.GetFiles(directoryToRemove, "*", SearchOption.AllDirectories).Any())
+                    {
+                        _fileSystem.DeleteDirectoryChecked(directoryToRemove, recursive: false, overrideAttributes: false, isSilent: true);
+                    }
+                }
+            }
+            else
+            {
+                foreach (var directoryToRemove in _fileSystem.GetDirectories(cacheLocation))
+                {
+                    _fileSystem.DeleteDirectoryChecked(directoryToRemove, recursive: true);
+                }
+            }
+
+            var filesAfterClean = _fileSystem.GetFiles(cacheLocation, "*.dat", SearchOption.AllDirectories);
+
+            if (configuration.CacheCommand.RemoveExpiredItemsOnly)
+            {
+                filesAfterClean = filesAfterClean.Where(f => _fileSystem.GetFileModifiedDate(f) < expirationTimer);
+
+                this.Log().Info("Removed {0} expired cached items in '{1}'", beforeFilesCount - filesAfterClean.Count(), cacheLocation);
+            }
+            else
+            {
+                this.Log().Info("Removed {0} cached items in '{1}'", beforeFilesCount - filesAfterClean.Count(), cacheLocation);
+            }
+        }
+
+        protected override string GetCommandDescription(CommandForAttribute attribute, ChocolateyConfiguration configuration)
+        {
+            return @"Get the statistics of what Chocolatey has cached, or clear any cached
+items in the current context.
+
+This is helpful for when it is necessary for the user to figure out
+when the correct version of a package is not installed, but instead a
+different version is used.";

```

Comment:
```suggestion
This is helpful for when it is necessary for the user to determine
when the correct version of a package is not installed, but instead a
different version is used.";

```

Reply:
I've read this several times and I don't understand what it means. Can it be rewritten?

Reply:
Looking over it again, I have no idea what I was trying to write out here.

Reply:
I have replaced this section with something else instead.